moped-editor/src/components/GridTable/Filters.js

```diff
@@ -481,103 +273,26 @@ const Filters = ({
   }, [setSearchParams, setFilters, setIsOr]);
 
-  /**
-   * Returns an array of strings containing messages about the filters.
-   * Returns null if no problems were found.
-   * @return {[]|null}
-   */
-  const handleApplyValidation = () => {
-    let feedback = [];
-    if (filterParameters) {
-      if (Object.keys(filterParameters).length === 0) {
-        feedback.push("• No filters have been added.");
-      } else {
-        Object.keys(filterParameters).forEach((filterKey) => {
-          const { field, value, gqlOperator } = filterParameters[filterKey];
-          if (field === null) {
-            feedback.push("• One or more fields have not been selected.");
-          }
-
-          if (gqlOperator === null) {
-            feedback.push("• One or more operators have not been selected.");
-          }
-
-          if (value === null || value.trim() === "") {
-            if (gqlOperator && !gqlOperator.includes("is_null")) {
-              feedback.push("• One or more missing values.");
-            }
-          }
-          if (!checkIsValidInput(filterParameters[filterKey])) {
-            feedback.push("• One or more invalid inputs.");
-          }
-        });
-      }
-    }
-    return feedback.length > 0 ? feedback : null;
-  };
-
   /**
    * Applies the current local state and updates the parent's state
    */
   const handleApplyButtonClick = () => {
-    const searchParamsFromFilters =
-      makeSearchParamsFromFilterParameters(filterParameters);
     setSearchParams((prevSearchParams) => {
-      const jsonParamString = JSON.stringify(searchParamsFromFilters);
+      const jsonParamString = JSON.stringify(filterParameters);
 
       prevSearchParams.set(advancedSearchFilterParamName, jsonParamString);
       prevSearchParams.set(advancedSearchIsOrParamName, isOrToggleValue);
       return prevSearchParams;
     });
 
     setIsOr(isOrToggleValue);
-    setFilters(searchParamsFromFilters);
+    setFilters(filterParameters);
     handleAdvancedSearchClose();
     // Clear simple search field in UI and state since we are using advanced search
     setSearchFieldValue("");
     setSearchTerm("");
   };
 
-  /**
-   * It returns true if the operator is null
-   * @param {object} field - The field being checked
-   * @returns {boolean}
-   */
-  const isFilterNullType = (field) => {
-    return field.gqlOperator && field.gqlOperator.includes("is_null");
-  };
-
-  /**
-   * This side effect monitors the count of filters
-   * if the count of filters is zero, then resets the state
-   */
-  useEffect(() => {
-    if (Object.keys(filterParameters).length === 0) {
-      setFilters([]);
-      handleClearFilters();
-      // Add an empty filter so the user doesn't have to click the 'add filter' button
-      generateEmptyFilter();
-    }
-  }, [filterParameters, setFilters, generateEmptyFilter, handleClearFilters]);
-
-  /**
-   * This side effect monitors whether the user has added a complete filter
-   */
-  useEffect(() => {
```

Comment:
The `filterComplete` state is removed and replaced with the `areAllFiltersComplete` helper. We can use that function to check if all filters are complete during render since it is calculated from `filterParameters`.

Reply:
This is interesting - i think that you might be on to something. I fixed this by adding the two operators that this filters uses ("subprojects_array_is_null" and "subprojects_array_is_not_null") to the list the code looks at when determining whether to show the value input or not (`OPERATORS_WITHOUT_SEARCH_VALUES`). Somehow the prod code handles this case without this change. 🤔

Reply:
Added "string_is_blank" and "string_is_not_blank" to that list as well to fix broken **Status is blank** or **Status is not blank** 😑